modules/placement-driver/src/main/java/org/apache/ignite/internal/placementdriver/PlacementDriverManager.java

```diff
@@ -187,10 +335,264 @@ private void takeOverActiveActor() {
      */
     private void stepDownActiveActor() {
         isActiveActor = false;
+
+        stopUpdater();
     }
 
     @TestOnly
     boolean isActiveActor() {
         return isActiveActor;
     }
+
+    /**
+     * Starts a dedicated thread to renew or assign leases.
+     */
+    private void startUpdater() {
+        //TODO: IGNITE-18879 Implement lease maintenance.
+        updater = new Thread(NamedThreadFactory.threadPrefix(clusterService.topologyService().localMember().name(), "lease-updater")) {
+            @Override
+            public void run() {
+                while (isActiveActor) {
+                    for (Map.Entry<ReplicationGroupId, Set<Assignment>> entry : groupAssignments.entrySet()) {
+                        ReplicationGroupId grpId = entry.getKey();
+
+                        ReplicationGroupLease lease = replicationGroups.getOrDefault(grpId, new ReplicationGroupLease());
+
+                        HybridTimestamp now = clock.now();
+
+                        if (lease.stopLeas == null || now.getPhysical() > (lease.stopLeas.getPhysical() - LEASE_PERIOD / 2)) {
+                            var leaseKey = ByteArray.fromString(PLACEMENTDRIVER_PREFIX + grpId);
+
+                            var newTs = new HybridTimestamp(now.getPhysical() + LEASE_PERIOD, 0);
+
+                            ClusterNode holder = nextLeaseHolder(entry.getValue());
```

Comment:
why the new leaseholder is selected each time the lease is going to expire?

Reply:
It can be needed, but make the selection more reasonable is a task of the maintenance ticket.